src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo;
+  onCloseModal: (value: null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, onCloseModal }) => {
+  const [loaded, setLoaded] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const { userId, id, title, completed } = selectedTodo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      const loadedUser = getUser(userId);
+
+      setUser(await loadedUser);
+      setLoaded(false);
+    };
+
+    loadUser();
+  }, [userId]);
+
+  const handleButtonClose = useCallback(
```

Comment:
The naming convention for event handler functions should follow the pattern `handle[EventName]`. Consider renaming `handleButtonClose` to `handleCloseButtonClick` to better align with the naming conventions.